### ISSUE TITLE
If training on a dataset using 320x320 image size, do I need to change in initialize biases function in yolo.py ?

### ISSUE THREAD

**Issue by LuthraBhomik** [ultralytics/yolov5#10228]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

If I am training on a dataset by resizing the images to 320x320, instead of 640x640, I would like to know if I need to change this line  **b.data[:, 4] += math.log(8 / (640 / s) ** 2)** in following piece of code in yolo.py.

It would be great if you give me an insight of this function.

 def _initialize_biases(self, cf=None):  # initialize biases into Detect(), cf is class frequency
        # https://arxiv.org/abs/1708.02002 section 3.3
        # cf = torch.bincount(torch.tensor(np.concatenate(dataset.labels, 0)[:, 0]).long(), minlength=nc) + 1.
        m = self.model[-1]  # Detect() module
        for mi, s in zip(m.m, m.stride):  # from
            b = mi.bias.view(m.na, -1)  # conv.bias(255) to (3,85)
            b.data[:, 4] += math.log(8 / (640 / s) ** 2)  # obj (8 objects per 640 image)
            b.data[:, 5:5 + m.nc] += math.log(0.6 / (m.nc - 0.99999)) if cf is None else torch.log(cf / cf.sum())  # cls
            mi.bias = torch.nn.Parameter(b.view(-1), requires_grad=True)

**Comment by glenn-jocher** [comment]
@LuthraBhomik yes you can change this, but this only affects bias initialization, so it won't have a huge impact on your final results.

**Comment by LuthraBhomik** [comment]
@glenn-jocher Can you please share the link to the formula/concept that is applied here. I would like to understand it.

**Comment by glenn-jocher** [comment]
@LuthraBhomik it's in the comment, equation is for an obj probability at stride s on a 640 image with an average of 8 objects per image.
